files/build_templates/docker_image_ctl.j2

```diff
@@ -76,36 +76,41 @@ function postStartAction()
         link_namespace $DEV
     fi
     # Wait until redis starts
-    /usr/bin/docker exec database ping_pong_db_insts
+    /usr/bin/docker exec database$DEV ping_pong_db_insts
     if [[ ("$BOOT_TYPE" == "warm" || "$BOOT_TYPE" == "fastfast") && -f $WARM_DIR/dump.rdb ]]; then
         rm -f $WARM_DIR/dump.rdb
     else
         # If there is a config db dump file, load it
-        if [ -r /etc/sonic/config_db.json ]; then
-            sonic-cfggen -j /etc/sonic/config_db.json --write-to-db
+        if [ -r /etc/sonic/config_db$DEV.json ]; then
+            sonic-netns-exec "$NET_NS" sonic-cfggen -j /etc/sonic/config_db$DEV.json --write-to-db
         fi
 
         if [[ "$BOOT_TYPE" == "fast" ]]; then
             # set the key to expire in 3 minutes
-            sonic-db-cli STATE_DB SET "FAST_REBOOT|system" "1" "EX" "180"
+            /usr/bin/sonic-netns-exec "$NET_NS" sonic-db-cli STATE_DB SET "FAST_REBOOT|system" "1" "EX" "180"
         fi
 
-        sonic-db-cli CONFIG_DB SET "CONFIG_DB_INITIALIZED" "1"
+        /usr/bin/sonic-netns-exec "$NET_NS" sonic-db-cli CONFIG_DB SET "CONFIG_DB_INITIALIZED" "1"
     fi
 
     if [[ -x /usr/bin/db_migrator.py ]]; then
         # Migrate the DB to the latest schema version if needed
         /usr/bin/db_migrator.py -o migrate
+
+        # Save in memory config_db to config_db.json for 2 reasons:
+        # 1. Persist the DB migration result.
+        # 2. Save in memory DB after warm reboot.
+        /usr/bin/config save -y
```

Comment:
How does it play out with config-setup?

config-setup.service is starting after database.service. When a switch boots for the first time the initial configuration will be generated by config-setup and loaded into the DB. With this change, an empty initial configuration will be created before config-setup has a chance to generate a proper config.

Reply:
Removed this line. We should not be doing config save here.